### DISCUSSION TITLE
[Questions] how to get rabbitmq metrics queue wise?

### DISCUSSION THREAD

**Discussion by amolngt** [rabbitmq/rabbitmq-server#12900] [Questions] [answered]
### Community Support Policy

- [X] I have read [RabbitMQ's Community Support Policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md)
- [X] I agree to provide all relevant information (versions, logs, rabbitmq-diagnostics output, detailed reproduction steps)

### RabbitMQ version used

3.13.7 or older

### Erlang version used

26.0.x

### Operating system (distribution) used

ubuntu

### How is RabbitMQ deployed?

Community Docker image

### rabbitmq-diagnostics status output

See https://www.rabbitmq.com/docs/cli to learn how to use rabbitmq-diagnostics
<details>

```
# PASTE OUTPUT HERE, BETWEEN BACKTICKS
```
</details>

### Logs from node 1 (with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs
<details>

```
# PASTE LOG HERE, BETWEEN BACKTICKS
```
</details>

### Logs from node 2 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs
<details>

```
# PASTE LOG HERE, BETWEEN BACKTICKS
```
</details>

### Logs from node 3 (if applicable, with sensitive values edited out)

See https://www.rabbitmq.com/docs/logging to learn how to collect logs
<details>

```
# PASTE LOG HERE, BETWEEN BACKTICKS
```
</details>

### rabbitmq.conf

See https://www.rabbitmq.com/docs/configure#config-location to learn how to find rabbitmq.conf file location
<details>

```
# PASTE rabbitmq.conf HERE, BETWEEN BACKTICKS
```
</details>

### Steps to deploy RabbitMQ cluster

currently i am working on my local machine. so no cluster.

### Steps to reproduce the behavior in question

1. run rabbitmq
2. publish some data
3. try to stop acknowledging messages
4. so now you will get unacknowledged messages

### advanced.config

See https://www.rabbitmq.com/docs/configure#config-location to learn how to find advanced.config file location
<details>

```
# PASTE advanced.config HERE, BETWEEN BACKTICKS
```
</details>

### Application code

<details>

```python
# PASTE CODE HERE, BETWEEN BACKTICKS
```
</details>

### Kubernetes deployment file

<details>

```yaml
# Relevant parts of K8S deployment that demonstrate how RabbitMQ is deployed
# PASTE YAML HERE, BETWEEN BACKTICKS
```
</details>

### What problem are you trying to solve?

Hi all,

RabbitMQ 3.9.13
Erlang 24.2.1

I want rabbitmq metrics queue wise. I tried using rabbitmq promethues plugin, which gives me metrics for example **"rabbitmq_queue_messages_unacked"**, but it is not queue wise. Let me know if you need more details. I want this to create an alerts if **rabbitmq_queue_messages_unacked** messages goes above some specific thresholds.

Thank you in advance.

**Comment by gomoripeti** [comment]
Hi

First of all please upgrade RabbitMQ from the very old and officially not supported 3.9. At least since version 3.10 RabbitMQ has a prometheus plugin bundled. Check out the documentation of the `/metrics/per-object` endpoint which has per queue metrics. https://www.rabbitmq.com/docs/prometheus#per-object-endpoint. (With recent RabbitMQ versions I'd suggest to use the `/metrics/detailed` endpoint which also has per queue metrics but supports selecting only the "metric families" you need, this way reducing the overhead, size, etc of returning the metrics https://www.rabbitmq.com/docs/prometheus#detailed-endpoint)

**Reply by amolngt** [reply]
Hi sir,
Thank you for the reply.

**Comment by michaelklishin** [comment, marked as answer]
1. With Prometheus https://www.rabbitmq.com/docs/prometheus#metric-aggregation
2. `GET /api/queues/{vhost}/{name}` (this is from memory, see HTTP API docs at `/api` on any node with the management plugin

Absolutely NOT to do: `GET /api/queues` to then get one metric of one queue. That will render 70 metrics per queue, so, 70K rows for 1000 queues.

**Reply by amolngt** [reply]
Hi sir,
Thank you for the reply.
